src/sentry/models/authidentity.py

```diff
@@ -22,6 +27,18 @@ class AuthIdentity(Model):
     last_synced = models.DateTimeField(default=timezone.now)
     date_added = models.DateTimeField(default=timezone.now)
 
+    def outbox_region_names(self) -> Collection[str]:
+        return find_regions_for_orgs([self.auth_provider.organization_id])
+
+    def handle_async_replication(self, region_name: str, shard_identifier: int) -> None:
```

Comment:
And here is the part where you can place the receiver logic directly inline with your model!  It also handles tombstones for free and focuses exclusively on the important happy path (you have a valid model and you're doing something with it).